robottelo/containers.py

```diff
@@ -0,0 +1,249 @@
+        is required.
+        Logs RHSM logs to stout.
+        Logs host name to stout after starting the container.
+        """
+        logging.info(f"Creating {self.tag} container named {self.name}")
+        volumes = (
+            {"/dev/log": {"bind": "/dev/log", "mode": "rw"}} if self._mount
+            else {}
+        )
+        self._inst = self._client.create_container(
+            detach=False,
+            host_config=self._client.create_host_config(
+                binds=volumes, port_bindings=self.ports
+            ),
+            image=f"{self.image}:{self.tag}",
+            ports=list(self.ports.keys()),
+            command="tail -f /var/log/rhsm/rhsm.log",
+        )
+        self._client.start(container=self._inst)
+        self.name = self.execute("hostname").strip()
+
+    def delete(self):
+        """Deletes all the containers belonging to an instance of this
+        class."""
+        logger.info(f"Deleting {self.name}")
+        self._client.remove_container(self._inst, v=True, force=True)
+
+    def execute(self, command):
+        """Executes a shell command.
+        Parameters:
+            command (init): A command to pass to the shell (CLI).
+        """
+        exec_inst = self._client.exec_create(
+            container=self._inst, cmd=command, stdout=True
+        )
+        logger.info(f"{self.name} is executing: {command}")
+        return self._client.exec_start(exec_id=exec_inst).decode()
+
+    def logs(self, file="stdout", tailing=False):
+        """Enables logging to a file. Defaults to writing to stdout.
+        Parameters:
+            file (str): A file name to write to.
+            tailing (boolian): If set to True, tails the log file.
+        """
+        if file == "stdout":
+            current = self._client.logs(self._inst["Id"])
+        else:
+            current = self.execute(f"cat {file}")
+        if isinstance(current, bytes):
+            current = current.decode()
+        self._logs[file] = current
+        return current.replace(
+            self._logs.get(file, ""), "") if tailing else current
+
+    def port(self):
+        """The port to use for SSH."""
+        return self._client.port(self._inst["Id"], 22)[0]["HostPort"]
+
+
+class Container:
+    """Generic class to gather information for a Linux Container.
+    It enables separating generic and runtime specific code.
+    Calls a runtime specific class to build the container. Defaults to docker.
+    Attributes:
+        runtime (init): The Linux container runtime to use. Defaults to docker.
+        image (init): The name of the container image to build from.
+        tag (init): The tag of the operating system to use.
+        agent (boolian): If set to True, prepares container for katello-agent.
+        port (init): The port to use for SSH.
+    Returns:
+        A container object
+    """
+
+    def __init__(
+            self, runtime="docker", image="ch-d", tag="rhel7", agent=False,
```

Comment:
why do you want to keep `tag="rhel7"` ? what if we wants to spin for `rhel6/8`?

Reply:
Vijay, this is normal Python function with normal default value. If you want rhel6 or rhel8, you just pass this string as argument value (e.g. `Container(tag="rhel8")`).